ctapipe/io/tests/test_table_loader.py

```diff
@@ -114,20 +118,42 @@ def test_true_parameters(test_file):
 
 def test_read_subarray_events(test_file_dl2):
     """Test reading subarray events"""
-
     from ctapipe.io.tableloader import TableLoader
 
     _, dl2_file = test_file_dl2
 
-    with TableLoader(
-        dl2_file, load_dl2=True, load_simulated=True, load_trigger=True
-    ) as table_loader:
+    with TableLoader(dl2_file, load_dl2=True, load_simulated=True) as table_loader:
         table = table_loader.read_subarray_events()
         assert "HillasReconstructor_alt" in table.colnames
         assert "true_energy" in table.colnames
         assert "time" in table.colnames
 
 
+def test_table_loader_keeps_original_order(dl2_merged_file):
+    """Test reading subarray events keeps order in file"""
+    from ctapipe.io.tableloader import TableLoader
+
+    # check that the order is the same as in the file itself
+    trigger = read_table(dl2_merged_file, "/dl1/event/subarray/trigger")
+    # check we actually have unsorted input
+    assert not np.all(np.diff(trigger["obs_id"]) >= 0)
```

Comment:
The input could be sorted there, e.g.`[0, 4 ,10]`
I suppose you are assuming obs_id ranges from 0 to len(trigger)?

Reply:
> The input could be sorted there, e.g. [0, 4 ,10]

Yes, and then the test wouldn't actually test that order is maintained. So the check here is to make sure the test is actually testing what it is supposed to test